Support M1 MacBooks darwin/arm64

<!-- Please only use this template for submitting enhancement requests -->

**What would you like to be added**:
When trying to build kubectl from tip (#0a839c6c3b9489716e216b4fc9dd2f9c17948c16) on a M1 Macbook:
```
$ make WHAT=cmd/kubectl
+++ [1228 14:35:00] Building go targets for darwin/arm64:
    ./vendor/k8s.io/code-generator/cmd/prerelease-lifecycle-gen
touch: _output/bin/prerelease-lifecycle-gen: No such file or directory
make[1]: *** [_output/bin/prerelease-lifecycle-gen] Error 1
make: *** [generated_files] Error 2
```

**Why is this needed**:
M1 MacBooks are a new major client platform that many developers will use. brew listed kubectl (kubernetes-cli) is one of the major packages still not supported on M1 (https://github.com/Homebrew/brew/issues/10152)
Since it's supported by go, it's only required to add it to the supported client list

Locally I managed to add support using the following patch:
```patch
diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index bef1d837..154eecfd 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -49,6 +49,7 @@ readonly KUBE_SUPPORTED_CLIENT_PLATFORMS=(
   linux/s390x
   linux/ppc64le
   darwin/amd64
+  darwin/arm64
   windows/amd64
   windows/386
 )
```

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -49,6 +49,7 @@ readonly KUBE_SUPPORTED_CLIENT_PLATFORMS=(
   linux/s390x
   linux/ppc64le
   darwin/amd64
+  darwin/arm64
   windows/amd64
   windows/386
 )
@@ -62,6 +63,7 @@ readonly KUBE_SUPPORTED_TEST_PLATFORMS=(
   linux/s390x
   linux/ppc64le
   darwin/amd64
+  darwin/arm64
   windows/amd64
 )
 
@@ -206,25 +208,30 @@ kube::golang::setup_platforms() {
     readonly KUBE_CLIENT_PLATFORMS
 
   elif [[ "${KUBE_FASTBUILD:-}" == "true" ]]; then
-    KUBE_SERVER_PLATFORMS=(linux/amd64)
+    host_arch=$(kube::util::host_arch)
+    if [[ "${host_arch}" != "amd64" && "${host_arch}" != "arm64" ]]; then
+      # on any platform other than amd64 and arm64, we just default to amd64
+      host_arch="amd64"
+    fi
+    KUBE_SERVER_PLATFORMS=("linux/${host_arch}")
     readonly KUBE_SERVER_PLATFORMS
-    KUBE_NODE_PLATFORMS=(linux/amd64)
+    KUBE_NODE_PLATFORMS=("linux/${host_arch}")
     readonly KUBE_NODE_PLATFORMS
     if [[ "${KUBE_BUILDER_OS:-}" == "darwin"* ]]; then
       KUBE_TEST_PLATFORMS=(
-        darwin/amd64
-        linux/amd64
+        "darwin/${host_arch}"
+        "linux/${host_arch}"
       )
       readonly KUBE_TEST_PLATFORMS
       KUBE_CLIENT_PLATFORMS=(
-        darwin/amd64
-        linux/amd64
+        "darwin/${host_arch}"
+        "linux/${host_arch}"
       )
       readonly KUBE_CLIENT_PLATFORMS
     else
-      KUBE_TEST_PLATFORMS=(linux/amd64)
+      KUBE_TEST_PLATFORMS=("linux/${host_arch}")
       readonly KUBE_TEST_PLATFORMS
-      KUBE_CLIENT_PLATFORMS=(linux/amd64)
+      KUBE_CLIENT_PLATFORMS=("linux/${host_arch}")
       readonly KUBE_CLIENT_PLATFORMS
     fi
   else

--- a/test/typecheck/main.go
+++ b/test/typecheck/main.go
@@ -48,10 +48,10 @@ var (
 	// interesting OS-based errors happen earlier.
 	crossPlatforms = []string{
 		"linux/amd64", "windows/386",
-		"darwin/amd64", "linux/arm",
-		"linux/386", "windows/amd64",
-		"linux/arm64", "linux/ppc64le",
-		"linux/s390x",
+		"darwin/amd64", "darwin/arm64",
+		"linux/arm", "linux/386",
+		"windows/amd64", "linux/arm64",
+		"linux/ppc64le", "linux/s390x",
 	}
 
 	// directories we always ignore

